### PR TITLE
WIP: cre technical patterns

### PR DESCRIPTION
**Pull request by Ottovsky**
Please treat it as an invitation to the discussion in comments, I think there is still some work to be done before merging. 

Let's agree whether those patterns fit the CRE idea and we could possibly support them in the future. Also if you have any comments regarding content, formatting, typos, please don't hesitate to raise them.

### CODE REVIEW

**Comment by darren-reddick** [comment]
Could we put a guide on running this locally? I've been running jekyll locally and its nice to have the option to see the changes rendered.

**Comment by cewood** [comment]
> Could we put a guide on running this locally? I've been running jekyll locally and its nice to have the option to see the changes rendered.

Maybe we could target migrating this to Hugo later on as well, if anyone is interested? There's no reason to be migrating sites to Jekyll in 2020, just more cargo culting :man_facepalming:

**Comment by darren-reddick** [comment]
> Maybe we could target migrating this to Hugo later on as well, 

Yes. That could be a good thing to do. 
How is this thing actually hosted atm anyway?

**Comment by cewood** [comment]
> > Maybe we could target migrating this to Hugo later on as well,
> 
> Yes. That could be a good thing to do.

Kewl, well we can add an item to our backlog if there's interest to track doing that later.


> How is this thing actually hosted atm anyway?

It's being published to GitHub pages, the hint is here https://github.com/ContainerSolutions/cloud-native-patterns/blob/master/CNAME

That could also be something we considering changing, moving it to Netlify or similar, so we can get viewable deployment previews on PR's, to properly see things before merging or releasing, etc. I've had great success using Cloudflare, Netlify, and Hugo before, super fast free hosting and nice experience overall.

**Comment by Ottovsky** [comment]
> > Could we put a guide on running this locally? I've been running jekyll locally and its nice to have the option to see the changes rendered.
> 
> Maybe we could target migrating this to Hugo later on as well, if anyone is interested? There's no reason to be migrating sites to Jekyll in 2020, just more cargo culting man_facepalming

I like the idea, I will create a low priority task in the jira for it.

File: _technical/cicd-as-platform.md
Comment (darren-reddick) [inline _technical/cicd-as-platform.md:28]: Having the word "product" here is confusing. Would it not read better if it just said "create a dedicated team" as a reference to the CI/CD Platfrom team.

File: _technical/gitops.md
Comment (darren-reddick) [inline _technical/gitops.md]: I dont see an InfrastructureAsCode pattern anywhere in this repo. I am going to remove this reference for now....

File: technical.md
Comment (darren-reddick) [inline technical.md]: I have tried to add a description but its a work in progress (and def better than "yadayada").

Would def appreciate input from others to try and get it to a sentence that encompasses what these patterns are about.

**Comment by darren-reddick** [comment]
Im not sold on the "structured-logging" pattern - i'm not sure what it is advocating: is it the centralization of the logs? Is it a particular format for the logs? Of course these are good things but I dont think the case for them is well presented here. 
The other patterns present a "scenario > problem > solution" structure which I think is missing here so I think we have more work to do on this one if we are going to keep it.

**Comment by Ottovsky** [comment]
Agree, I was not bought by this when I refactored it. I think it is about having logs structured and serialized in a way that it is easy to store and query in ELK or something similar. The reference to the RFC is not very clear to me, I would just drop it for now.

File: _technical/cicd-as-platform.md
Comment (Ottovsky) [reply _technical/cicd-as-platform.md:28]: You are right, without defining what does the "product team" means, we should not use it here.

File: _technical/structured-logging.md
Comment (Ottovsky) [inline _technical/structured-logging.md]: As per @darren-reddick comment, what does the rest of the @ContainerSolutions/cre thinks about this technical pattern? Should we try to refactor it or just get rid of it for the time being?

File: CONTRIBUTING.md
Comment (darren-reddick) [inline CONTRIBUTING.md]: I think this should actually be <context> as mentioned in the Confluence doc

File: CONTRIBUTING.md
Comment (darren-reddick) [inline CONTRIBUTING.md]: This section actually describes the problems and pitfalls which are found within this context. So maybe <problems within context>?

**Comment by darren-reddick** [comment]
> I totally forgot about that :( I will adjust it accordingly

Here you go.... https://container-solutions.atlassian.net/wiki/spaces/CRE/pages/263356540/Tech+Patterns+Cloud+Native+Patterns+Discussion

File: technical.md
Comment (cewood) [reply technical.md]: Looks pretty close to me. Is this an improvement?

```suggestion
description: "Patterns that focus on the technical and architectural aspects of how to build and operate Cloud Native services whilst ensuring their reliability and security."
```

File: _technical/structured-logging.md
Comment (cewood) [reply _technical/structured-logging.md]: Structured logging is definitely legitimate thing, and refers to the tokenisation of logs. So in terms of standard fields, and also in terms of formatting, whether that be JSON or comma separated with a fixed field order, etc...

File: technical.md
Comment (darren-reddick) [reply technical.md]: Thanks. That is better now.

File: _technical/structured-logging.md
Comment (darren-reddick) [reply _technical/structured-logging.md]: Yes. I dont think the argument is against the pattern but that it isnt presented in a particularly clear way here (for me anyway). I think this needs a fair bit of work to get into a better state.

File: CONTRIBUTING.md
Comment (rodrmartinez) [inline CONTRIBUTING.md]: 2. If you've added a  pattern, double check for typos

File: CONTRIBUTING.md
Comment (rodrmartinez) [inline CONTRIBUTING.md]: If you've changed a pattern, explain your changes with a comment in the PR

File: _technical/cicd-as-platform.md
Comment (rodrmartinez) [inline _technical/cicd-as-platform.md]: This means feature requests should be phrased in a way that is useful to

File: _technical/dynamic-secrets.md
Comment (rodrmartinez) [inline _technical/dynamic-secrets.md]: - All credentials are managed (manually) be a security team. Nobody usually knows where this team actually sits, the only interface is an obscure ticket filing system.

File: _technical/ephemeral-environments.md
Comment (rodrmartinez) [inline _technical/ephemeral-environments.md:13]: Container technology has taken us a long way to resolve the 'it works on my machine' problem. However, in a microservice environment, the number of dependencies for truly validating the functionality of a service can still make this a big challenge.

**Review by rodrmartinez — COMMENTED** [review]
Just some typos

File: _technical/ephemeral-environments.md
Comment (darren-reddick) [reply _technical/ephemeral-environments.md:13]: I think you are commenting on the code pre-change here.

File: _technical/structured-logging.md
Comment (darren-reddick) [reply _technical/structured-logging.md]: I have opened an issue to track this [here](https://github.com/ContainerSolutions/cloud-native-patterns/issues/10) and will remove the pattern from this PR.

**Review by cewood — APPROVED** [review]
LGTM :+1: 

@darren-reddick can you please review and merge if there's no further feedback.

File: _technical/cicd-as-platform.md
Comment (darren-reddick) [inline _technical/cicd-as-platform.md]: The image should be the first text after the meta-data section at the top to follow the standard pattern. This applies to all of the patterns.

**Review by darren-reddick — CHANGES_REQUESTED** [review]

File: _technical/cicd-as-platform.md
Comment (darren-reddick) [reply _technical/cicd-as-platform.md]: Screenshot 2020-12-14 at 15 25 59

**Review by darren-reddick — APPROVED** [review]

**Comment by darren-reddick** [comment]
Have run it using a jekyll server locally and LGTM

**Pull request merged**
